File: selectionSort.cpp
#include<iostream>
using namespace std;

int main(){

  int arr[20],size;

  cin>>size;

for(int i=0;i<size;i++){
  cout<<"a["<<i<<"]:";
  cin>>arr[i];
}

int temp=arr[0],pos;

for(int i=0;i<size;i++){

  pos=i;
  temp=arr[i];
  for(int j=i;j<size;j++){
    if(temp<arr[j]){
      temp=arr[j];
      pos=j;
    }
  }
  int var;
  var=arr[pos];
  arr[pos]=arr[i];
  arr[i]=var;

}

for(int i=0;i<size;i++){
  cout<<arr[i]<<" ";
}

}

File: sorting/selectionSort.java
class Main{

  int arr[] = {4,3,9,6,7};

  void sort(){

    for(int i=0 ;i<arr.length ;i++){

      int tempMax = arr[i];
      int index = i;
      for(int j=i+1 ;j<arr.length ;j++){

        if(tempMax<arr[j]){
          tempMax = arr[j];
          index = j;
        }
      }

      int temp = arr[index];
      arr[index] = arr[i];
      arr[i] = temp;
    }

  }

  void print(){
    for(int i =0;i<arr.length ;i++){
      System.out.print(arr[i]+" ");
    }
  }

  public static void main(String[] args) {

    Main obj = new Main();
    obj.sort();
    obj.print();
  }
}
